src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -27,18 +27,13 @@ namespace OpenTelemetry.Metrics
 {
     internal sealed class MeterProviderSdk : MeterProvider
     {
-        internal const int MaxMetrics = 1000;
         internal int ShutdownCount;
-        private readonly Metric[] metrics;
-        private readonly Metric[] metricsCurrentBatch;
         private readonly List<object> instrumentations = new List<object>();
         private readonly List<Func<Instrument, MetricStreamConfiguration>> viewConfigs;
         private readonly object collectLock = new object();
-        private readonly object instrumentCreationLock = new object();
-        private readonly HashSet<string> metricStreamNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
         private readonly MeterListener listener;
         private readonly MetricReader reader;
-        private int metricIndex = -1;
+        private readonly CompositeMetricReader compositeMetricReader;
```

Comment:
We need to dispose this as well?

Reply:
We have a `this.reader?.Dispose()` which should take care of it. I have defensively added a `this.compositeReader.Dispose()` as well now.